src/imex/import/useImportTable.utils.ts

```diff
@@ -179,37 +181,44 @@ export const parseRawData = async <D extends { id?: string | number }>(
     const rawRowValues = Object.values(row)
     for (let index = 0; index < rawRowValues.length; index++) {
       const rawCell = rawRowValues[index]
-      if (!orderedColumns[index]) {
+      const currentColumn = orderedColumns[index]
+      if (!currentColumn) {
         continue
       }
 
+      const columnDataPath = (
+        typeof currentColumn.Header === 'string'
+          ? currentColumn.Header
+          : currentColumn.imex?.path
+      ) as string
+
       let cellError: string | null = null
 
       const parse = (value: any) =>
-        orderedColumns[index]?.imex?.parse?.(value, row) ?? value
+        currentColumn.imex?.parse?.(value, row) ?? value
 
       let newCellValue: string | number | string[] | number[] | undefined =
         rawCell
 
-      const ignoreEmpty = orderedColumns[index]?.imex?.ignoreEmpty ?? true
+      const ignoreEmpty = currentColumn.imex?.ignoreEmpty ?? true
 
       try {
         newCellValue = parseCell(
           rawCell,
-          orderedColumns[index]!.imex!.type as IMEXColumnType,
+          currentColumn.imex?.type as IMEXColumnType,
           { parse, ignoreEmpty }
         )
 
         // If parsed value is null, throw if required and ignore if not.
         if (newCellValue == null) {
-          if (orderedColumns[index]?.imex?.required) {
+          if (currentColumn.imex?.required) {
             throw new Error(ParsingErrors[ParseCellError.REQUIRED])
           } else if (ignoreEmpty) {
             continue
           }
         }
 
-        const validate = orderedColumns[index]?.imex?.validate ?? (() => true)
+        const validate = currentColumn.imex?.validate ?? (() => true)
         const validateResponse = validate(newCellValue, row)
```

Comment:
It would be nice to also have the parsed row and not just the raw values.

Reply:
Hum. It would been to add a loop for every value. Is it necessary somewhere ? Maybe we can wait to need it ?

Reply:
It becomes useful whenever the validation of one column depends on another. The last example I can remember is the import of booking contacts made by @Carolinevp.

You only need 2 loops:
- one to parse all raw values first
- another to validate the columns against the parsed and raw values

Reply:
https://habxfr.atlassian.net/browse/APP-26382